### ISSUE TITLE
Primary site routes for categories accessible across all sites/locales

### ISSUE THREAD

**Issue by chasegiunta** [craftcms/cms#3796]
### Description
Easiest way to explain is through describing the current situation:

There's a category titled "Accessibility" in English, with the slug of `accessibility`.
French: "Tests d'accessibilité" & `tests-accessibilite`.
Spanish: "Pruebas de accesibilidad" & `pruebas-accesibilidad`
German: "Accessibility Testing" & `accessibility-testing`

They all live in a "Resources" category group.

![image](https://user-images.githubusercontent.com/1377169/52381412-ec259f80-2a36-11e9-9169-0506edfee1d5.png)

When visiting:
```
domain.test/es/resources/accessibility
domain.test/fr/resources/accessibility
domain.test/de/resources/accessibility
```

All of these  _should_ 404, but they don't. They all render the `resources/_category` template, and the inherited `category` variable is that from the English category version.

This is causing Google to index the incorrect pages. 

`category.title` & `category.slug` in the template both return English variations ("Accessibility" `accessibility`), while `craft.app.language` & `currentSite.language` are both actually returning their respective languages (es, fr, or de) so Craft is aware of the language - it just seems to be ignoring the URI prefixes upon routing categories.

### Additional info

- Craft version: 3.1.8
- PHP version: 7.1.24
- Database driver & version: MariaDB (MySQL 5.5.5)
- Plugins & versions: n/a

**Comment by brandonkelly** [comment]
What are your sites’ base URLs set to?

**Comment by chasegiunta** [comment]
We just inherited this site, and they're set to
```
domain.test
domain.test/de/
domain.test/fr/
domain.test/es/
```
I know on past Craft 2 sites we've set up the index.php in folders for each language. Is that how this should be set up?

**Comment by brandonkelly** [comment]
Nope, no need to make subfolders anymore.

Can you throw this into the template and see what it outputs?

```twig
{{ currentSite.name }}
```

Make sure it’s showing the expected site name on each of the domains.

**Comment by chasegiunta** [comment]
Yep! So, I mentioned trying  `{{ currentSite.language }}` in the original issue description, but just went ahead & attempted `{{ currentSite.name }}` on the template that's rendered for `/fr/resources/accessibility` which, theoretically, should be 404'ing. It outputs French. Same with `/de/resources/accessibility` for German and `/es/resources/accessibility` for Spanish. All outputting the correct site name.

**Comment by brandonkelly** [comment]
Huh… are there any custom routes defined in Settings → Routes, or `config/routes.php` ?

**Comment by chasegiunta** [comment]
No. I was thinking maybe those too initially. There was actually an entries section titled, “Resources”, as well with the `resources` handle & slug, so I nuked that section locally just to ensure there wasn’t something funky or conflicting going on. Same issue. Perhaps there’s some task that could regenerate these category routes?

I should emphasize this is only happening with categories, not entries.

Will try tomorrow to recreate with a new install to rule out any other unknown variables.

**Comment by brandonkelly** [comment]
Yeah not sure what else could be the culprit… can you just send in a ticket from the Craft Support widget on Dashboard? Can try to reproduce it with your files from there.

**Comment by chasegiunta** [comment]
Just to update, I ended up creating a bare test case with a new Craft install and couldn't reproduce this. No idea what is going on, and I was able to write a quick twig fix for the project to redirect to the correct locale. If I have a few spare minutes I'll dupe the repo and see if I can wittle down the whole project to eliminate various factors that could be potential causes — but for now I'll close until I can do that and provide more details.
